### ISSUE TITLE
MA0040 shouldn't trigger on overloads marked with [Experimental]

### ISSUE THREAD

**Issue by abatishchev** [meziantou/Meziantou.Analyzer#1156]
### Version of the Meziantou.Analyzer NuGet package

3.0.79

### Rule Identifier

MA00400

### Target Framework

net10.0

### C# Language version

C# 13.0

### Description

MA00400 shouldn't be triggered by the existence of an overload which accepts `CancellationToken` if such an overload is experimental, i.e. decorated with `System.Diagnostics.CodeAnalysis.ExperimentalAttribute', such as:

```csharp
namespace Microsoft.Extensions.Http.Resilience;

public static class HttpClientResiliencePredicates
{
  public static bool IsTransient(Outcome<HttpResponseMessage> outcome);

  [Experimental("EXTEXP0001", UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}")]
  public static bool IsTransient(
    Outcome<HttpResponseMessage> outcome,
    CancellationToken cancellationToken);
}
```

See [IsTransient(Outcome<HttpResponseMessage>, CancellationToken)](ttps://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.http.resilience.httpclientresiliencepredicates.istransient?view=net-10.0-pp#microsoft-extensions-http-resilience-httpclientresiliencepredicates-istransient(polly-outcome((system-net-http-httpresponsemessage))-system-threading-cancellationtoken))

### Reproduction Steps

The following code:

```c#
services.AddHttpClient("MyClient")
        .AddStandardResilienceHandler(static options =>
             options.Retry.ShouldHandle = static args => ValueTask.FromResult(
                 args.Outcome.Result?.StatusCode == HttpStatusCode.Conflict ||
                 HttpClientResiliencePredicates.IsTransient(args.Outcome))) // <-- this method has an experimental oveload
```

Triggers
